sycl/unittests/Extensions/CommandGraph/Exceptions.cpp

```diff
@@ -919,21 +919,15 @@ TEST_F(CommandGraphTest, AsyncAllocKindExceptionCheck) {
 
   Graph.begin_recording(Queue);
 
-  addAsyncAlloc<OperationPath::RecordReplay, usm::alloc::host>(Graph, Queue,
-                                                               1024, Ptr1);
```

Comment:
Can you clarify if this PR is adding host-allocation support to Graphs for just regular queue submission?
If it's not adding functionality to SYCL-Graph, then I think we should still be throwing an error here because it remains unsupported. However, if we are adding functionality then we need tests in https://github.com/intel/llvm/tree/sycl/sycl/test-e2e/Graph/AsyncAlloc for that

Reply:
Yeah in this PR it's going to bypass the graph memory pool even if the host allocation is being recorded which is not going to work (if we had tests for it). We should be going through the graph path for all allocations and let the graph memory pool throw an error if we don't support it (it already does this [here](https://github.com/intel/llvm/blob/80b06481ca7e5012ab687d0d53115df5e5a855af/sycl/source/detail/graph_memory_pool.cpp#L74)).

Reply:
Reverted this test changes and reverted the graph async alloc path